aesara/tensor/shape.py

```diff
@@ -462,13 +462,9 @@ def connection_pattern(self, node):
     def grad(self, inp, grads):
         x, *shape = inp
         (gz,) = grads
-        # Should I set an SpecifyShape on gz? I think so
-        # But I don't do it now as we need to make an optimization
-        # to remove that op from the graph to don't block other optimization
-        # Should I do an optimizer that will remove the SpecifyShape?
-        # I think Yes
-        # return [specify_shape(gz, s)] + [aesara.gradient.DisconnectedType()() for _ in range(len(shape))]
-        return [gz] + [aesara.gradient.DisconnectedType()() for _ in range(len(shape))]
+        return [specify_shape(gz, shape)] + [
```

Comment:
If anyone can help designing a test that covers this change, that would be great. This fixes an immediate failure when creating a grad graph in `test_batch_normalization_train`, but that test is incredibly convoluted.